aviary/subsystems/energy/battery_builder.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import openmdao.api as om
+
+from aviary.subsystems.subsystem_builder_base import SubsystemBuilderBase
+from aviary.subsystems.energy.battery_sizing import SizeBattery
+from aviary.variable_info.variables import Aircraft, Dynamic
+
+
+class BatteryBuilder(SubsystemBuilderBase):
+    def build_pre_mission(self, aviary_inputs=None):
+        return SizeBattery(aviary_inputs=aviary_inputs)
+
+    def get_mass_names(self):
+        return [Aircraft.Battery.MASS]
+
+    def build_mission(self, num_nodes, aviary_inputs=None) -> om.Group:
+        battery_group = om.Group()
+        soc = om.ExecComp('state_of_charge = (energy_capacity - (electric_energy/efficiency)) / energy_capacity',
+                          state_of_charge={'val': np.zeros(
+                              num_nodes), 'units': 'unitless'},
+                          energy_capacity={'val': 10.0, 'units': 'kJ'},
+                          electric_energy={'val': np.zeros(num_nodes), 'units': 'kJ'},
+                          efficiency={'val': 0.95, 'units': 'unitless'})
+
+        battery_group.add_subsystem('state_of_charge',
```

Comment:
Currently it looks like SoC is getting calculated but never constrained - is the expectation that users must always manually specify they want that constraint active, rather than having the subsystem builder set it up?

Reply:
Good call, I've changed it so battery state of charge is constrained by default.